pkg/smbios/smbios_legacy_test.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2016-2021 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package smbios
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/u-root/u-root/pkg/memio"
+)
+
+var tmpBuf = []byte{0, 0, 0, 0, 0, 0}
+
+func mockMemioRead(base int64, uintn memio.UintN) error {
```

Comment:
I haven't written a lot of mocked/dependency-injected tests in u-root like this, usually I've relied on doing a QEMU-based VM test. I've probably been wrong not to, and I definitely want to keep these. However, I'd also like a QEMU-based VM test with a test for this legacy reader.

At some point, we should probably run all our VM tests with a variety of firmware (seabios, coreboot, UEFI OVMF), but for now, we just do Seabios, so only the legacy will work in a VM.

If you just add a test which [is skipped for UID != 0](https://github.com/u-root/u-root/blob/11adccf0a563b7c5baee0aa2a7a90b6724709cf0/pkg/testutil/testutil.go#L168), that will run as a generic VM test under https://github.com/u-root/u-root/blob/master/integration/gotests/gotest_test.go and be covered.

Reply:
Done.